sysutils/osxbom/Portfile

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+# My repo is ahead of the main one in terms of versions:
+github.setup        cooljeanius osxbom 0.0.5
+github.tarball_from archive
+categories          sysutils
+maintainers         {gwmail.gwu.edu:egall @cooljeanius} openmaintainer
+platforms           darwin
+license             public-domain
+
+description         A free re-implementation of the NextSTEP/OSX lsbom \
+                    utility
+long_description    ${description}. Used by the PureDarwin project \
+                    to work with MacPorts binary packages.
+homepage            https://cauldrondevelopment.com/svn/${name}/trunk/
+
+checksums           rmd160  ae9fa1120123aa0f19d1a92bce869d530556bf19 \
+                    sha256  788c6650494eff481baebe1129335003301d9530c20ad61cb0ec013b628797ca \
+                    size    155958
+
+# probably not actually necessary to autoreconf, but whatever:
+use_autoreconf      yes
+autoreconf.args     -fvi -Wall
+
+# mostly just things picked up by trace mode:
+depends_build-append \
+                    port:gawk \
+                    port:grep \
+                    bin:nroff:groff \
+                    bin:gmkdir:coreutils \
+                    port:libmacho-headers
+depends_lib-append  port:libunwind
+
+set CFLAGS          "${configure.cflags} [get_canonical_archflags cc]"
+
+build.env-append    CC=${configure.cc} \
+                    CXX=${configure.cxx} \
+                    CPP=${configure.cpp}
+
+pre-build {
+    reinplace "s|-Werror|${CFLAGS}|" ${worksrcpath}/Makefile
+}
+
+test.run            yes
+test.target         check
+test.target-append  distcheck
+depends_test-append bin:gzip:gzip \
+                    bin:zip:zip
```

Comment:
Distchecks are not necessary.

Reply:
But I want to include it.

Reply:
While you're the main maintainer, ports are required to be open (with `openmaintainer` being in the `maintainers` field), so others are compelled to run tests should they also decide to update the port.

The `distcheck` target doesn't test the functionality of the project without needing to rebuilding the project, since it creates an entire tarball for distribution; this is only helpful when creating releases for the project, not when testing functionality alone. The `check` target achieves what the `test` fields are meant to do.

Reply:
it's also helpful in ensuring you haven't screwed up the build system, which I consider to be an important part of port functionality

Reply:
Tests are more focused on testing the functionality of the project, not testing whether files trivial of a distribution have been updated. This is ultimately handled by upstream; in other words, upstream has to make sure that distribution files are up-to-date on top testing functionality.

In this case, testing whether `lsbom` functions as intended is more of MacPorts's concern in relation to what the `test` fields are designed to do. MacPorts is (to some extent) obligated to distribute distribution files (e.g `NEWS`), but in no way, shape, or form should MacPorts be concerned whether these files have the necessary content pertaining to a specific version of the project. That should be upstream's concern, given that they're also responsible for releasing the project and its release-related files (e.g tarballs).

The `check` target covers the concerns of  MacPorts, testing the functionality of  the `lsbom` binary, while the `distcheck` target covers the concerns of upstream.

Reply:
ok, well, I am also upstream, and I want to test these things in MacPorts, too. I can remove the openmantainer line if other people are concerned about running it.

Reply:
Accepting the port would be halted by the need of a good reason for why you alone can update any one port.

In recognizing you being upstream, you should handle any concerns that the `distcheck` addresses on your own, rather than adding that responsibility to MacPorts, which you already do, given the project this port represents has its own CI.

Reply:
Removed `openmaintainer` since apparently it is too much of a hassle for other people for the `distcheck` target to be part of the optional `test` target that no one even has to run anyways.